src/client/GraphQLClient/errorHandlers.ts

```
@@ -1,26 +1,21 @@
 import { IntegrationLogger } from '@jupiterone/integration-sdk-core';
 import { AttemptContext, sleep } from '@lifeomic/attempt';
+import { GraphQlQueryResponse } from '@octokit/graphql/dist-types/types';
+import { GraphqlResponseError } from '@octokit/graphql';
 
-type GraphQLError = {
-  type?: 'NOT_FOUND' | 'FORBIDDEN' | 'RATE_LIMITED' | string;
-  path?: string[];
-  locations?: { line: number; column: number }[];
-  extensions?: {
-    code: string;
-    typeName: string;
-    fieldName: string;
-  };
-  message: string;
-  documentation_url?: string;
-};
-
-const handleTypeErrors = (errors, logger, type: string): boolean => {
-  errors = Array.isArray(errors) ? errors : [errors];
-
-  if (errors?.every((error) => error.type === type)) {
+const handleTypeErrors = (
+  errors: GraphQlQueryResponse<never>['errors'],
+  logger,
+  type: string,
+): boolean => {
+  if (Array.isArray(errors) && errors?.every((error) => error.type === type)) {
+    logger.info(
```

Could you add a comment here explaining what this is for? If every error is `type` we ignore them?

---

Yeah, this method is generic, below there are two cases in which we ignore the errors. 1. if the type = FORBIDDEN 2. if the type = NOT_FOUND.

They are safe to ignore and allow the ingestion to continue.

---

I'll add a comment

---

I can see why we'd ignore a NOT_FOUND error, but is there any concern for inadvertently masking missing ingesting data because our credentials aren't given enough permissions and repeatedly getting FORBIDDEN?

I'm not saying I think this code is incorrect as written.  I just wonder if we might want to put in a TODO to address the FORBIDDEN errors once we determine how we're going to report skipped items to users.

---

The reporting feature would be great here.
In the errors I have seen, this comes up for only a portion of the response data, a nest part of the graph query. I'll get a todo in there